autoPlay.user.js

```diff
@@ -691,11 +702,11 @@ function useMetalDetectorIfRelevant() {
 		var enemy = g_Minigame.m_CurrentScene.GetEnemy(g_Minigame.m_CurrentScene.m_rgPlayerData.current_lane, g_Minigame.m_CurrentScene.m_rgPlayerData.target);
 		// check if current target is a boss, otherwise we won't use metal detector
 		if (enemy && enemy.m_data.type == ENEMY_TYPE.BOSS) {
-			var enemyBossHealthPercent = enemy.m_flDisplayedHP / enemy.m_data.max_hp;
+			var secsToBossDeath = predictBossDeath();
 
-			if (enemyBossHealthPercent >= 0.9) { // We want sufficient time for the metal detector to be applicable
+			if (secsToBossDeath <= 10) { // We want sufficient time for the metal detector to be applicable
```

Comment:
Maybe 
    if (secsToBossDeath <= 10 && secsToBossDeath > 1)
will be sufficient